core/src/services/libsql/backend.rs

```diff
@@ -0,0 +1,345 @@
+        let query = format!(
+            "SELECT {} FROM {} WHERE `{}` = ? LIMIT 1",
+            self.value_field, self.table, self.key_field
+        );
+        let client = SyncClient::from_config(self.get_config()?).map_err(|err| {
+            Error::new(ErrorKind::Unexpected, "connection failed").set_source(err)
+        })?;
+        let rs = client
+            .execute(Statement::with_args(query, args!(path)))
+            .map_err(|err| Error::new(ErrorKind::Unexpected, "get failed").set_source(err))?;
+        let val = rs.rows.first().map(|row| row.values.get(0));
+        match val {
+            Some(Some(v)) => match v {
+                Value::Null => Ok(None),
+                Value::Blob { value } => Ok(Some(value.to_owned())),
+                _ => Err(Error::new(ErrorKind::Unexpected, "invalid value type")),
+            },
+            _ => Ok(None),
+        }
+    }
+
+    async fn set(&self, path: &str, value: &[u8]) -> Result<()> {
+        let cloned_path = path.to_string();
+        let cloned_value = value.to_vec();
+        let cloned_self = self.clone();
+
+        task::spawn_blocking(move || cloned_self.blocking_set(cloned_path.as_str(), &cloned_value))
+            .await
+            .map_err(|err| {
+                Error::new(
+                    ErrorKind::Unexpected,
+                    "unhandled error from libsql when spawning task",
+                )
+                .set_source(err)
+            })
+            .and_then(|inner_result| inner_result)
+    }
+
+    fn blocking_set(&self, path: &str, value: &[u8]) -> Result<()> {
+        let query = format!(
+            "INSERT OR REPLACE INTO `{}` (`{}`, `{}`) VALUES (?, ?)",
+            self.table, self.key_field, self.value_field
+        );
+        let client = SyncClient::from_config(self.get_config()?).map_err(|err| {
+            Error::new(ErrorKind::Unexpected, "connection failed").set_source(err)
+        })?;
+
+        client
+            .execute(Statement::with_args(query, args!(path, value.to_vec())))
+            .map_err(|err| Error::new(ErrorKind::Unexpected, "set failed").set_source(err))?;
+        Ok(())
+    }
+
+    async fn delete(&self, path: &str) -> Result<()> {
+        let cloned_path = path.to_string();
+        let cloned_self = self.clone();
+
+        task::spawn_blocking(move || cloned_self.blocking_delete(cloned_path.as_str()))
+            .await
+            .map_err(|err| {
+                Error::new(
+                    ErrorKind::Unexpected,
+                    "unhandled error from libsql when spawning task",
+                )
+                .set_source(err)
+            })
+            .and_then(|inner_result| inner_result)
+    }
+
+    fn blocking_delete(&self, path: &str) -> Result<()> {
+        let query = format!("DELETE FROM {} WHERE `{}` = ?", self.table, self.key_field);
+
+        let client = SyncClient::from_config(self.get_config()?).map_err(|err| {
```

Comment:
We need to handle libsql's errors properly. In the case of a "database locked" error, we can set it as temporary and allow opendal to retry it.